File: Python/62-二叉搜索树的第k个结点.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回对应节点TreeNode
    def KthNode(self, pRoot, k):
        # write code here
        lst = []
        def helper(root):
            if not root:
                return
            helper(root.left)
            lst.append(root)
            helper(root.right)
        helper(pRoot)
        if 0 < k <= len(lst):
            return lst[k - 1]
        return None
        

File: C++/62-二叉搜索树的第k个结点.cpp
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};
*/
class Solution
{
public:
    void helper(TreeNode *root, vector<TreeNode *> &vt)
    {
        if (!root)
            return;
        helper(root->left, vt);
        vt.push_back(root);
        helper(root->right, vt);
    }
    TreeNode *KthNode(TreeNode *pRoot, int k)
    {
        vector<TreeNode *> ret;
        helper(pRoot, ret);
        if (k <= 0 || k > ret.size())
            return NULL;
        return ret[k - 1];
    }
};